realm/realm-library/src/main/java/io/realm/BaseRealm.java

```diff
@@ -433,6 +438,36 @@ public void cancelTransaction() {
         sharedRealm.cancelTransaction();
     }
 
+    /**
+     * Returns a frozen snapshot of the current Realm. This Realm can be read and queried from any thread without throwing
+     * an {@link IllegalStateException}. A frozen Realm has its own lifecycle and can be closed by calling {@link #close()},
+     * but fully closing the Realm that spawned the frozen copy will also close the frozen Realm.
+     * <p>
+     * Frozen data can be queried as normal, but trying to mutate it in any way or attempting to register any listener will
+     * throw an {@link IllegalStateException}.
+     * <p>
+     * Note: Keeping a large number of Realms with different versions alive can have a negative impact on the filesize
+     * of the Realm. In order to catch such use patterns it is possible to set {@link RealmConfiguration.Builder#maxNumberOfActiveVersions(long)}.
```

Comment:
Shouldn't we make a default for maxNumberOfActiveVersions? This seems like something people are bound to accidentally do.

Reply:
The default is `Long.MAX_VALUE`, which is essentially the same as disabling. Finding a good default might be difficult as it very much depends on what else the app is doing, but I guess we cant set it to something relatively high and the make the exception message really clear about how to change the value if people hit it and think it is a mistake.